src/domain/usecases/UpdateHcvDataValuesUseCase.ts

```diff
@@ -0,0 +1,150 @@
+                csvPath: options.csvPath,
+                settings: settings,
+            });
+        }
+
+        if (options.post) {
+            logger.debug("Updating events...");
+            const result = await this.programEventsRepository.save(eventsToUpdate);
+            logger.debug("Events updated");
+
+            if (settings.dataElement.remove) {
+                const programStageToUpdate = this.removeParentDataElementFromProgramStage(
+                    programStage,
+                    settings
+                );
+                logger.debug(
+                    `Removing dataElement ${settings.dataElement.id} from program stage ${programStage.id}...`
+                );
+                await this.programStageRepository.save(programStageToUpdate);
+                logger.debug("DataElement removed");
+            }
+
+            return result;
+        } else {
+            return { type: "success" };
+        }
+    }
+
+    private removeParentDataElementFromProgramStage(
+        programStage: ProgramStage,
+        settings: HcvSettings
+    ): ProgramStage {
+        const programStageWithoutParentDataElement = _(programStage.dataElements)
+            .map((programStageDataElement): Maybe<ProgramStageDataElement> => {
+                if (programStageDataElement.dataElement.id === settings.dataElement.id) return undefined;
+                return programStageDataElement;
+            })
+            .compact()
+            .value();
+
+        const programStageToUpdate: ProgramStage = {
+            ...programStage,
+            dataElements: programStageWithoutParentDataElement,
+        };
+        return programStageToUpdate;
+    }
+
+    private eventsToUpdate(events: ProgramEvent[], settings: HcvSettings): ProgramEvent[] {
+        return _(events)
+            .map(event => {
+                const parentDataValue = event.dataValues.find(
+                    dataValue => dataValue.dataElementId === settings.dataElement.id
+                );
+
+                if (!parentDataValue) return undefined;
+
+                return {
+                    ...event,
+                    dataValues: this.checkDataValueCondition(event, settings, parentDataValue),
+                };
+            })
+            .compact()
+            .value();
+    }
+
+    private checkDataValueCondition(
+        event: ProgramEvent,
+        settings: HcvSettings,
+        parentDataValue: EventDataValue
+    ): EventDataValue[] {
+        const dataElementsToUpdate = _(settings.dataElement.dataElementsToUpdate)
+            .map((dataElementToUpdate): Maybe<EventDataValue> => {
+                return {
+                    dataElementId: dataElementToUpdate.id,
+                    value:
+                        dataElementToUpdate.condition === parentDataValue.value
+                            ? dataElementToUpdate.value
+                            : "",
+                    storedBy: "",
```

Comment:
It makes sense to keep the same storedBy than the original dataValue, right? (not sure if that's overriden by the user performing the post, though)